File: datasets/Python/Copilot/Algorithm/medium5.py
def compute_grundy(m, k):
    if m == 1:
        return 1
    depth = 0
    while m > 1:
        m = max(1, m // k)
        depth += 1
    return depth % 2

g = int(input())
for _ in range(g):
    n = int(input())
    xor_sum = 0
    for _ in range(n):
        m, k = map(int, input().split())
        xor_sum ^= compute_grundy(m, k)
    print("BOB" if xor_sum != 0 else "BEN")

File: datasets/C++/Copilot/Algorithm/medium5.cpp
#include <iostream>
using namespace std;

int computeGrundy(long long m, int k) {
    if (m == 1) return 1;
    int depth = 0;
    while (m > 1) {
        m = max(1LL, m / k);
        depth++;
    }
    return depth % 2;
}

int main() {
    int g;
    cin >> g;
    while (g--) {
        int n;
        cin >> n;
        int xor_sum = 0;
        for (int i = 0; i < n; ++i) {
            long long m;
            int k;
            cin >> m >> k;
            xor_sum ^= computeGrundy(m, k);
        }
        cout << (xor_sum ? "BOB" : "BEN") << endl;
    }
    return 0;
}
